nursery_store.go

```diff
@@ -0,0 +1,1377 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/boltdb/bolt"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+	"github.com/roasbeef/btcd/wire"
+)
+
+// NurseryStore abstracts the persistent storage layer for the utxo nursery.
+// Concretely, it stores commitment and htlc outputs that until any time-bounded
+// constraints have fully matured. The store exposes methods for enumerating
+// its contents, and persisting state transitions detected by the utxo nursery.
+type NurseryStore interface {
+
+	// Incubate registers a commitment output and a variadic number of htlc
+	// outputs to be swept back into the user's wallet. The event is
+	// persisted to disk, such that the nursery can resume the incubation
+	// process after a potential crash.
+	Incubate(*kidOutput, ...babyOutput) error
+
+	// CribToKinder atomically moves a babyOutput in the crib bucket to the
+	// kindergarten bucket. The now mature kidOutput contained in the
+	// babyOutput will be stored as it waits out the kidOutput's CSV delay.
+	CribToKinder(*babyOutput) error
+
+	// PreschoolToKinder atomically moves a kidOutput from the preschool
+	// bucket to the kindergarten bucket. This transition should be executed
+	// after receiving confirmation of the preschool output's commitment
+	// transaction.
+	PreschoolToKinder(*kidOutput) error
+
+	// AwardDiplomas accepts a variadic number of kidOutputs from the
+	// kindergarten bucket, and removes their corresponding entries from the
+	// height and channel indexes. If this method detects that all outputs
+	// for a particular contract have been incubated, it returns the channel
+	// points that are ready to be marked as fully closed.
+	// TODO: make this handle one output at a time?
+	AwardDiplomas(...kidOutput) error
```

Comment:
Any reason this is variadic instead of an array?

Reply:
Similar to above comment, will modify to make a slice.